mixer/pkg/config/crd/admit.go

```diff
@@ -360,28 +362,37 @@ func (ac *AdmissionController) admit(review *v1alpha1.AdmissionReview) *v1alpha1
 		}
 	}
 
-	var t store.ChangeType
+	if !watched(ac.options.ValidateNamespaces, review.Spec.Namespace) {
+		return &v1alpha1.AdmissionReviewStatus{Allowed: true}
+	}
+
+	ev := &store.BackendEvent{
+		Key: store.Key{
+			Namespace: review.Spec.Namespace,
+			Kind:      review.Spec.Kind.Kind,
+		},
+	}
 	switch review.Spec.Operation {
 	case admission.Create, admission.Update:
-		t = store.Update
+		ev.Type = store.Update
+		var obj unstructured.Unstructured
+		if err := yaml.Unmarshal(review.Spec.Object.Raw, &obj); err != nil {
+			return makeErrorStatus("cannot decode configuration: %v", err)
+		}
+		ev.Value = backEndResource(&obj)
+		ev.Key.Name = ev.Value.Metadata.Name
 	case admission.Delete:
-		t = store.Delete
+		if review.Spec.Name == "" {
+			return makeErrorStatus("illformed request: name not found on delete request")
+		}
+		ev.Type = store.Delete
+		ev.Key.Name = review.Spec.Name
 	default:
 		glog.Warningf("Unsupported webhook operation %v", review.Spec.Operation)
 		return &v1alpha1.AdmissionReviewStatus{Allowed: true}
 	}
 
-	var obj unstructured.Unstructured
-	if err := yaml.Unmarshal(review.Spec.Object.Raw, &obj); err != nil {
-		return makeErrorStatus("cannot decode configuration: %v", err)
-	}
-
-	if !watched(ac.options.ValidateNamespaces, obj.GetNamespace()) {
-		return &v1alpha1.AdmissionReviewStatus{Allowed: true}
-	}
-
-	ev := toEvent(t, &obj)
-	if err := ac.options.Validator.Validate(&ev); err != nil {
+	if err := ac.options.Validator.Validate(ev); err != nil {
```

Comment:
is this where `jsonpb.Parse` is called with the specific proto?

Reply:
yes, the implementation of `ac.options.Validator` will be https://github.com/istio/istio/blob/master/mixer/pkg/config/store/validator.go#L55, which conducts the `jsonpb.Parse`. Also, an upcoming PR will do further validation on converted proto objects.